collated/test/yeggasd.md

```diff
@@ -0,0 +1,635 @@
+        writer.write(TEST_DATA);
+        writer.close();
+
+        SecurityUtil.encrypt(TEST_DATA_FILE, hashedPassword);
+        thrown.expect(WrongPasswordException.class);
+        SecurityUtil.decrypt(TEST_DATA_FILE);
+    }
+
+    @Test
+    public void encryptDecrypt_wrongPassword_throwsWrongPasswordException() throws Exception {
+
+        SecurityUtil.encrypt(VALID_DATA_FILE, hashedPassword);
+        thrown.expect(WrongPasswordException.class);
+        SecurityUtil.decryptFile(VALID_DATA_FILE, new Password(WRONG_PASSWORD));
+    }
+
+    @Test
+    public void encryptDecrypt_wrongPasswordBadPadding_throwsWrongPasswordException() throws Exception {
+        FileWriter writer = new FileWriter(TEST_DATA_FILE);
+        writer.write(TEST_DATA);
+        writer.close();
+
+        SecurityUtil.encryptFile(TEST_DATA_FILE, new Password(TEST_PASSWORD));
+        thrown.expect(WrongPasswordException.class);
+        SecurityUtil.decryptFile(TEST_DATA_FILE, new Password(WRONG_PASSWORD));
+    }
+
+    @Test
+    public void encryptDecryptFile_wrongPassword_throwsWrongPasswordException() throws Exception {
+        byte[] hashedWrong = SecurityUtil.hashPassword(WRONG_PASSWORD);
+
+        FileWriter writer = new FileWriter(TEST_DATA_FILE);
+        writer.write(TEST_DATA);
+        writer.close();
+
+        SecurityUtil.encrypt(TEST_DATA_FILE, hashedPassword);
+        thrown.expect(WrongPasswordException.class);
+        SecurityUtil.decrypt(TEST_DATA_FILE, hashedWrong);
+    }
+
+    @Test
+    public void encrypt_wrongPasswordLength_throwsAssertionError() throws Exception {
+
+        FileWriter writer = new FileWriter(TEST_DATA_FILE);
+        writer.write(TEST_DATA);
+        writer.close();
+        byte[] truncatedHashedPassword = Arrays.copyOf(hashedPassword, 13);
+
+        thrown.expect(AssertionError.class);
+        SecurityUtil.encrypt(TEST_DATA_FILE, truncatedHashedPassword);
+    }
+
+    @Test
+    public void decrypt_wrongPasswordLength_throwsAssertionError() throws Exception {
+
+        FileWriter writer = new FileWriter(TEST_DATA_FILE);
+        writer.write(TEST_DATA);
+        writer.close();
+        byte[] truncatedHashedPassword = Arrays.copyOf(hashedPassword, 13);
+
+        thrown.expect(AssertionError.class);
+        SecurityUtil.decrypt(TEST_DATA_FILE, truncatedHashedPassword);
+    }
+
+    @After
+    public void reset() throws Exception {
+        SecurityUtil.decrypt(VALID_DATA_FILE, hashedPassword);
+    }
+}
+```
+###### /java/seedu/address/logic/parser/RemovePasswordCommandParserTest.java
+``` java
+/**
+ * As we are only doing white-box testing, our test cases do not cover path variations
+ * outside of the RemovePasswordCommand code. For example, inputs "1" and "1 abc" take the
+ * same path through the RemovePasswordCommand, and therefore we test only one of them.
+ * The path variation for those two cases occur inside the ParserUtil, and
+ * therefore should be covered by the ParserUtilTest.
+ */
```

Comment:
If you find yourself explaining the "why" behind something you're doing in the code, it may be more appropriate to shift the explanation to the real documentation documents (e.g. Developer Guide).

Reply:
Same below where you copied this comment